grpc.go

```diff
@@ -442,6 +447,9 @@ func (a *agentGRPC) updateContainerConfigNamespaces(config *configs.Config) erro
 	// If no NEWPID type was found, let's make sure we add it. Otherwise,
 	// the container could end up in the same PID namespace than the agent
 	// and we want to prevent this for security reasons.
+	// The container would then join the shared pid namespace of either the
+	// pause process created by the agent or the pid namespace of the infra
+	// container.
 	if !pidNs {
```

Comment:
Without NEWPID settings in the config, we should still let `ctr.useSandboxPidNs` decide if the container will join the sandbox level pidns. Otherwise `ctr.useSandboxPidNs` is not working if container spec has no NEWPID settings. The mix of sandbox pidns, container use sandbox pidns and NEWPID settings is complicating the protocol semantics. So I want to simplify it by moving NEWPID settings out of the picture.

Reply:
Typically, when `ctr.useSandboxPidNs` is set, NEWPID will be set because that bool indicates that indicates that the container needs to join a namespace belonging to the sandbox and this will be provided in the OCI spec file.
This is also in accordance with your earlier comment
`no NEWPID: the container joins the sandbox level pidns
NEWPID with empty pidns path: the container lives in its own pidns
NEWPID with non-empty pidns path: the container joins the sandbox level pidns`

no NEWPID-> container joins the sandbox level pidns (which is what is done here)

Reply:
My comment as you quoted above is to remove `ctr.useSandboxPidNs`. If you keep it, we should not rely on `NEWPID` settings here. The point is that we should not depend on two separate flags for one thing. 

> Typically, when ctr.useSandboxPidNs is set, NEWPID will be set

We cannot assume it. This is a protocol API and we need to make it clear what is in effect and what is not.